tests/unit/test_cpulinear.cpp

```diff
@@ -0,0 +1,72 @@
+#include <deepworks/deepworks.hpp>
+
+#include "kernels_reference.hpp"
+#include "test_utils.hpp"
+
+namespace dw = deepworks;
+
+struct CPULinearModelTest: public ::testing::Test {
+    CPULinearModelTest()
+        : in(dw::Shape{batch_size, in_features}),
+          model(in, dw::Linear(out_features, "linear")(in))  {
+        model.compile();
+    }
+
+    int in_features  = 64;
+    int out_features = 10;
+    int batch_size   = 32;
+
+    dw::Placeholder in;
+    dw::Model model;
+};
+
+TEST_F(CPULinearModelTest, CPULinearForward) {
+    dw::Tensor X_train(in.shape());
+    dw::initializer::uniform(X_train);
+
+    dw::Tensor actual(model.outputs()[0].shape());
+    model.forward(X_train, actual);
+
+    dw::Tensor expected(model.outputs()[0].shape());
+    auto W = model.layers()[0].params()[0].data();
+    auto b = model.layers()[0].params()[1].data();
```

Comment:
Why not?
```cpp
    auto W = model.layers()[0].params()['weights'].data();
    auto b = model.layers()[0].params()['bias'].data();
```

The use of indexes is not clear to the user and for layers with many trainable parameters (e.g. BatchNorm).

Reply:
Possible usage.
```
for (auto&& layer : model.layers()) {
    for (auto&& p : layer.params()) {
        p.train(false)
    }
}
```

I agree with you, it can be represented in several ways, but I'm going to add this in separate PR, to minimize changes